libnmstate/netapplier.py

```diff
@@ -120,19 +121,22 @@ def generate_ifaces_metadata(ifaces_desired_state, ifaces_current_state):
     )
 
 
-def sanitize_ethernet_state(ifaces_desired_state):
+def sanitize_ethernet_state(ifaces_desired_state, ifaces_current_state):
     """
     If auto-negotiation, speed and duplex settings are not provided, they need
     to be set to None to not override them with the values from the current
     settings since the current settings are read from the device state and not
     from the actual configuration.  This makes it possible to distiguish
     whether a user specified these values in the later configuration step.
     """
-    for iface_state in six.viewvalues(ifaces_desired_state):
-        ethernet = iface_state.setdefault("ethernet", {})
-        ethernet.setdefault("auto-negotiation", None)
-        ethernet.setdefault("speed", None)
-        ethernet.setdefault("duplex", None)
+    for ifname, iface_state in six.viewitems(ifaces_desired_state):
+        if ifname in ifaces_current_state:
+            iface_current_state = ifaces_current_state[ifname]
+            if iface_current_state['type'] == 'ethernet':
```

Comment:
IMHO it would be better to check if the current state contains an ethernet subtree here. The method to create the current state needs to be changed to only add the ethernet subtree if the interface type is ethernet.

Also wired.create_setting() should only be called if the interface type is ethernet.

Reply:
I will add the changes not to report the `ethernet` subtree, but see my previous response regarding the verification here.
Thanks for the two points.